src/pool/pool_scalable.c

```diff
@@ -264,15 +283,15 @@ static umf_result_t tbb_pool_initialize(umf_memory_provider_handle_t provider,
         return UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY;
     }
 
-    int ret = init_tbb_callbacks(&pool_data->tbb_callbacks);
+    int ret = init_tbb_callbacks();
```

Comment:
I just noted that we have a possible memory leak in this function (not related to this particular PR).

If the `init_tbb_callbacks()` fails the `tbb_pool_initialize` will exit at line 289 and the `pool_data` is leaked.
Could you please fix it in this PR as well or at least create an issue?

Reply:
👍 it is a one-line fix (just add `umf_ba_global_free(pool_data);` there), so it will be faster to fix it than to submit an issue ;-)

Reply:
Actually, it is a two-line fix :). There are two possible returns with errors.